### DISCUSSION TITLE
Hamburger in navigation bar is not appearing

### DISCUSSION THREAD

**Discussion by dvats** [alshedivat/al-folio#2472] [Q&A] [answered]
I hVe been using the al-folio theme for a few years now and only just realized that the hamburger (three horizontal lines for the navigation bar at the top) do not appear on small screens as they should. Even the contents of the top navigation bar disappear. Here is my website: https://dvats.github.io/

Is there a place where I can turn this on/off? There is a [discussion with a similar issue](https://github.com/alshedivat/al-folio/discussions/781), but the details on the answer there are not clear to me.

**Comment by george-gca** [comment, marked as answer]
Which discussion is that?

I believe this is a bug that has been fixed years ago, probably even before I started as a contributor here, since I've never seen that bug. I would recommend [updating your base code to the latest version](https://github.com/alshedivat/al-folio/blob/master/INSTALL.md#upgrading-from-a-previous-version) of our code, but I recognize that this would require some effort. If you would do that it would probably be easier to start from scratch and apply your changes on top of the latest version.

**Reply by dvats** [reply]
Sorry, this is the discussion: https://github.com/alshedivat/al-folio/discussions/781  (updated in the main post now)

Yes, one solution is to essentially restart, but I would like that to be my last resort!

**Reply by george-gca** [reply]
Try to change your bootstrap for [this one](https://getbootstrap.com/docs/4.6/getting-started/introduction/#bundle).

By reading that discussion it says that the person was using bootstrap without popper, which should be something like `bootstrap.min.js`. But the correct one is to use `bootstrap.bundle.min.js`. Also looking at the console in your site it seems that the jquery is not loaded? It says `ReferenceError: $ is not defined`, but maybe fixing the bootstrap fixes this.

**Reply by dvats** [reply]
Thanks George, and I apologize in advance for my ignorance on the technical aspects of this.

There is no `bootstrap.bundle.min.js` in my `_config.yml` or looks like anywhere. Where exactly should I look to make these changes?

**Reply by george-gca** [reply]
Previously I believe these were in `_includes/scripts/bootstrap.html`. But just search for `bootstrap` in your repository to check where your bootstrap js integrity is used.

**Reply by dvats** [reply]
The only place "bootstrap" appears in my repo is in `_table.scss` as a comment, and in the `_config.yml` file. Could you explain what I should change here:

```
bootstrap:
  version: "4.6.1"
  integrity:
    css: "sha256-DF7Zhf293AJxJNTmh5zhoYYIMs2oXitRfBjY+9L//AY="
    js: "sha256-fgLAgv7fyCGopR/gBNq2iW3ZKIdqIcyshnUULC4vex8="
bootstrap-table:
  version: "1.21.4"
```

**Reply by george-gca** [reply]
It would be easier if I could see your source code. Check where these values are used, search for `site.bootstrap.integrity.js` in your code. This is probably in the file `_includes/scripts/bootstrap.html`. It is probably something like this:

```
<script src="https://cdn.jsdelivr.net/npm/bootstrap@{{ site.bootstrap.version }}/dist/js/bootstrap.bundle.min.js" integrity="{{ site.bootstrap.integrity.js }}" crossorigin="anonymous"></script>
```

Check if in this part it states `bootstrap.bundle.min.js` or `bootstrap.min.js`.

**Reply by dvats** [reply]
There is no "scripts" folder in my `_includes` folder and there seems to be no file called `bootstrap.html`. I am sharing a copy of my code via the following temporary dropbox link: 
https://www.dropbox.com/scl/fo/cquggf6frye5no1yckjnp/AGAmWcs6Lwv0VEOFalmzejY?rlkey=xax4qxqahtzj7ypxjckdsqg4z&st=hff6onae&dl=0

I hope this helps, and thank you for being patient with me!

**Reply by george-gca** [reply]
It's like I said, your site is much older than I thought. And it is missing the bootstrap part of the code.

Try to include this line:
```
<script src="https://cdn.jsdelivr.net/npm/bootstrap@{{ site.bootstrap.version }}/dist/js/bootstrap.bundle.min.js" integrity="{{ site.bootstrap.integrity.js }}" crossorigin="anonymous"></script>
```

in your `_includes/hemline.html`. Also include

```
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/bootstrap@{{ site.bootstrap.version }}/dist/css/bootstrap.min.css" integrity="{{ site.bootstrap.integrity.css }}" crossorigin="anonymous">
```

in your `_includes/head.html`. See if that solves it.

**Reply by dvats** [reply]
I tried this and now the navigation bar appears like this, and there is not hamburger when the site is small either

<img width="400" alt="Screenshot 2024-06-03 at 9 39 00 AM" src="https://github.com/alshedivat/al-folio/assets/6656840/836b8fd1-d459-44c2-a049-317473995828">

When I remove the line added to `head.html`, things go back to normal, but still not hamburger.

**Reply by dvats** [reply]
I see that there is no `bootstrap.bundle.min.js` file in my `assets/js/` folder. Would that be an issue?

**Reply by george-gca** [reply]
Not really, since you added that line to your `_includes/hemline.html`. I am afraid that I can't help much, since there has been too many changes from your version to the current version (or even the one I started playing with). And looking again at your code, it is from a version before the hamburguer in navigation was implemented. For instance, [this is the code](https://github.com/alshedivat/al-folio/blob/b4f90ff416b5961136cb9ed59e4edbcdf02109c1/_includes/header.liquid#L27) responsible for the hamburguer button:

```html
      <!-- Navbar Toggle -->
      <button
        class="navbar-toggler collapsed ml-auto"
        type="button"
        data-toggle="collapse"
        data-target="#navbarNav"
        aria-controls="navbarNav"
        aria-expanded="false"
        aria-label="Toggle navigation"
      >
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar top-bar"></span>
        <span class="icon-bar middle-bar"></span>
        <span class="icon-bar bottom-bar"></span>
      </button>
```

And I don't see it in your code in `_includes/header.html`. One thing you can do is look for the specific commits in the repo that introduced that change and try to cherry pick them to your repository, but this probably will cause more trouble than starting fresh and applying your changes on top of the latest version.

**Reply by dvats** [reply]
Thank you for your time and effort. For now I would even be happy with ensuring that the navigations buttons on the top don't disappear on small screens, so users can go to other pages. Is there a way I could do that?

**Reply by george-gca** [reply]
Since this would involve designing and developing such solution, I believe the easiest ways would be the ones I suggested. Since your site doesn't seem to have much customization, only new content, I think it wouldn't be that hard to port your changes to the newest version. Besides you could benefit from the new code organization and features.

**Reply by dvats** [reply]
Thanks for your help and suggestions! I have now moved to the new code and all seems to work.
